pkg/config/config.go

```diff
@@ -18,10 +17,29 @@ var (
 		"instance-url",
 		field.WithDescription(`The GitHub instance URL to connect to. (default "https://github.com")`),
 	)
+	appIDField = field.StringField(
+		"app-id",
+		field.WithDescription("The GitHub App to connect to."),
+	)
+
+	appPrivateKeyPath = field.StringField(
+		"app-privatekey-path",
+		field.WithDescription("Path to private key that is used to connect to the GitHub App"),
+	)
 	syncSecrets = field.BoolField(
 		"sync-secrets",
 		field.WithDescription(`Whether to sync secrets or not`),
 	)
+	fieldRelationships = []field.SchemaFieldRelationship{
+		field.FieldsMutuallyExclusive(
```

Comment:
maybe we should also add `FieldsAtLeastOneUsed(accessTokenField, appPrivateKeyPath)` since we removed the `field.WithRequired(true)` from `accessTokenField`